edgelet/iotedge/src/check/checks/aziot_edged_version.rs

```diff
@@ -18,8 +24,113 @@ impl Checker for AziotEdgedVersion {
     fn description(&self) -> &'static str {
         "latest security daemon"
     }
-    fn execute(&mut self, check: &mut Check, _: &mut tokio::runtime::Runtime) -> CheckResult {
-        self.inner_execute(check)
+    fn execute(
+        &mut self,
+        check: &mut Check,
+        tokio_runtime: &mut tokio::runtime::Runtime,
+    ) -> CheckResult {
+        let settings = if let Some(settings) = &check.settings {
+            settings
+        } else {
+            return CheckResult::Skipped;
+        };
+
+        if settings.parent_hostname().is_some() {
+            return CheckResult::Ignored;
+        }
+
+        let latest_versions = if let Some(expected_aziot_edged_version) =
+            &check.expected_aziot_edged_version
+        {
+            future::Either::A(future::ok::<_, Error>(crate::LatestVersions {
+                iotedged: expected_aziot_edged_version.clone(),
+            }))
+        } else {
+            let proxy = std::env::var("HTTPS_PROXY")
+                .ok()
+                .or_else(|| std::env::var("https_proxy").ok())
+                .map(|proxy| proxy.parse::<hyper::Uri>())
+                .transpose()
+                .context(ErrorKind::FetchLatestVersions(
+                    FetchLatestVersionsReason::CreateClient,
+                ));
+            let hyper_client = proxy.and_then(|proxy| {
+                MaybeProxyClient::new(proxy, None, None).context(ErrorKind::FetchLatestVersions(
+                    FetchLatestVersionsReason::CreateClient,
+                ))
+            });
+            let hyper_client = match hyper_client {
+                Ok(hyper_client) => hyper_client,
+                Err(err) => return CheckResult::Failed(err.into()),
+            };
+
+            let request = hyper::Request::get("https://aka.ms/latest-iotedge-stable")
+                .body(hyper::Body::default())
+                .expect("can't fail to create request");
+
+            future::Either::B(
+                hyper_client
+                    .call(request)
+                    .then(|response| -> Result<_, Error> {
+                        let response = response.context(ErrorKind::FetchLatestVersions(
+                            FetchLatestVersionsReason::GetResponse,
+                        ))?;
+                        Ok(response)
+                    })
+                    .and_then(move |response| match response.status() {
+                        hyper::StatusCode::MOVED_PERMANENTLY => {
```

Comment:
I know this is the original code, but this will fail if aka.ms starts returning 3xx instead of 301 specifically. (I'm surprised it returns 301 instead of 302, actually, given that aka.ms redirects are mutable.)

Change it to:

```rust
if response.status().is_redirection() {
    ...
}
else {
    ...
}
```